Arm/MinTheory.lean

```diff
@@ -8,10 +8,10 @@ attribute [minimal_theory] ite_false
 attribute [minimal_theory] dite_true
 attribute [minimal_theory] dite_false
 attribute [minimal_theory] ite_self
-attribute [minimal_theory] and_true
-attribute [minimal_theory] true_and
-attribute [minimal_theory] and_false
-attribute [minimal_theory] false_and
+attribute [minimal_theory high] and_true
```

Comment:
@alexkeizer A comment explaining why these rules are cheap, and thus can have a high priority, would be welcome.

Reply:
I added the following:
```
/-
Notice how both `and_true : ?p ∧ True = ?p` and `and_self : ?p ∧ ?p = ?p` may
be attempted by `simp` on a goal of the shape `_ ∧ True`, as they are both
a match where the discrimination tree is concerned.

However, assuming the left conjunct is not def-eq to `True`, an attempt of
`and_self` will fail to unify, which causes a fallback to `and_true`.
For the latter simp lemma, on the other hand, we know that if the discrimination
tree gives a match, then the lemma should be applicable.
This is because the variable `?p` is used only once in the pattern (i.e.,
the pattern is linear), and the first unification of an unassigned metavariable
is always successful.

Another possibly expensive lemma is
  `and_iff_left_of_imp : {a b : Prop} (ha : a → b) : a ∧ b ↔ a`
The pattern of this lemma (`?a ∧ ?b`) is perfectly linear, but to apply such
a lemma, `simp` first has to discharge the `(ha : a → b)`, which might fail,
and might be expensive.

Thus, the rationale we follow is that only linear, side-condition free lemmas
get the `high` priority, and everything else gets the default prioriry.
This ensures that `and_true` gets tried before `and_self` or
`and_iff_left_of_imp`
-/
```

I then also went through the rest of the file and mechanically applied this heuristic.

This seems to not have had much of an impact on the heartbeat usage, but still feels like morally the right thing to do. Experiments do show that some part of the LNSym standard simpsets is rather expensive, even if we don't use anything.